app/controllers/posts_controller.rb

```diff
@@ -8,5 +8,22 @@ def show
     @user = User.find(params[:user_id])
     @post = @user.posts.find(params[:id])
     @post_counter = params[:post_counter]
+    @comment = Comment.new
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    current_user = User.find(params[:user_id])
```

Comment:
- [ ] As of now the current user is the first user just as provided by the application controller. Therefore, there is no need to find the current_user.

  Look at the requirements below:
  
  - > Creates a Post on behalf of the current_user.
  
  Please refactor as necessary.

Reply:
Thanks for pointing this out, I got pumped and implemented the new functionalities for every user.

😄